drivers/can/Kconfig.stm32fd

```
@@ -0,0 +1,40 @@
+# STM32 CAN configuration options
+
+DT_COMPAT_STM32_FDCAN := st,stm32-fdcan
+
+config CAN_STM32FD
+	bool
+	default $(dt_compat_enabled,$(DT_COMPAT_STM32_FDCAN))
+	select CAN_MCAN
+	select USE_STM32_LL_RCC
+
+if CAN_STM32FD
+
+config CAN_MAX_STD_ID_FILTER
+	int "Maximum number of std ID filters"
+	default 28
+	range 0 28
+	help
+	  Defines the maximum number of filters with standard ID (11-bit)
+	  that can be attached.
+
+config CAN_MAX_EXT_ID_FILTER
+	int "Maximum number of ext ID filters"
+	default 8
+	range 0 8
+	help
+	  Defines the maximum number of filters with extended ID (29-bit)
+	  that can be attached.
+
+config CAN_CKDIV
+	int "CKDIV register value"
+	range 0 15
+	default 0
+	help
+	  This value is written to the CKDIV register.
+	  The APB clock is divided according to this value before it is feed to
+	  CAN core. Note that the the divider affects all CAN controllers.
+	  The values of the register are multiplied by two and zero corresponds
+	  to one. The value six, for example results in a clock divided by 12.
```

I did not understand this explanation without looking into the manual, so I suggest below change and using `>> 2` in the code to get the value actually written to the register. This is also more consistent with other clock settings via Kconfig.

```suggestion
config CAN_STM32_CLOCK_DIVISOR
	int "CAN clock divisor"
	range 1 30
	default 1
	help
	  The APB clock is divided by this value (stored in CKDIV register)
	  before it is fed to the CAN core.
	  Note that the the divisor affects all CAN controllers.

	  Allowed values: 1 or 2 * n with n <= 15.
```

---

IMO it is better if we don't allow invalid settings in the first place. Maybe I can come up with a better help text.

---

I think it's most important that Kconfig settings can be easily understood without knowing specific details of the underlying hardware.

It's very common in Zephyr to only mention the allowed values for clock divisors in the help text if it's not possible to check for correct values via Kconfig directly, see e.g. [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/clock_control/Kconfig.stm32g4).

---

Changed. I added a compile time check to guarantee the n*2

---

Nice, the compile-time check is a good idea!